include/grpc/event_engine/event_engine.h

```diff
@@ -321,17 +321,17 @@ class EventEngine : public std::enable_shared_from_this<EventEngine> {
   /// called.
   virtual bool CancelConnect(ConnectionHandle handle) = 0;
   /// Provides asynchronous resolution.
+  ///
+  /// This object has a destruction-is-cancellation semantic.
+  /// Implementations should make sure that all pending requests are cancelled
+  /// when the object is destroyed and all pending callbacks will be called
+  /// shortly.
+  ///
+  /// Note: in the gRPC's async implementation, cancellation is "best effort" in
```

Comment:
Keep in mind that this is a public contract, and as such, changing it as little as possible will be the best thing for our customers. It's not appropriate to suggest internal implementation details in this contract, since they may change.

You can say something like "If cancellation races with request completion, implementations may choose to either cancel or satisfy the request."

Reply:
Good point. Done.